### PR TITLE
Add InvalidProxyConfigurationWarning for HTTPS proxies

### PR DESCRIPTION
**Pull request by sethmlarson**
I pre-allocated #1850 for users that are receiving the issue. I have a feeling we may need to lock the conversation if it attracts too much traffic but it's better than leaving users without an official means of solving the issue.

cc @jalopezsilva

### CODE REVIEW

**Comment by joelverhagen** [comment]
Oh no, what have I done?

**Comment by sethmlarson** [comment]
@joelverhagen My apologies, I meant @jalopezsilva :) But I'm sure you're up to no good! ;)

File: src/urllib3/poolmanager.py [hunk 1]
```diff
@@ -459,9 +461,21 @@ def _set_proxy_headers(self, url, headers=None):
             headers_.update(headers)
         return headers_
 
+    def _validate_proxy_scheme_url_selection(self, url_scheme):
+        if url_scheme == "https" and self.proxy.scheme == "https":
+            warnings.warn(
```
Comment (pquentin) [inline src/urllib3/poolmanager.py:466]: Is there a sensible stacklevel to use here?

Comment (sethmlarson) [reply src/urllib3/poolmanager.py:466]: Probably! I'll add one.

**Comment by sethmlarson** [comment]
I like your wording more @pquentin, was struggling to come up with something late last night. :)

File: src/urllib3/poolmanager.py (hunk 1 above)
Comment (sethmlarson) [reply src/urllib3/poolmanager.py:466]: Settled with 3 so the traceback lands on wherever `urlopen()` is called. There's not really a great way to do [multi-entrypoint warning stacklevel](https://twitter.com/sethmlarson/status/1245074746645057542). :(

Comment (pquentin) [reply src/urllib3/poolmanager.py:466]: Yeah, that tweet is how I found out about stacklevel. In my case I'm using `urllib3.ProxyManager` and stacklevel=5 ends up in my code

**Pull request merged**